src/types.h

```diff
@@ -240,13 +240,24 @@ class SizedType {
     return size_bits_ / 8;
   }
 
-  void SetSize(size_t size)
+  void SetSize(size_t byte_size)
   {
-    size_bits_ = size * 8;
-    if (IsIntTy()) {
-      assert(size == 0 || size == 1 || size == 8 || size == 16 || size == 32 ||
-             size == 64);
-    }
+    if (IsIntTy())
+      SetIntBitWidth(byte_size * 8);
+    else
+      size_bits_ = byte_size * 8;
+  }
+
+  void SetIntBitWidth(size_t bits)
+  {
+    assert(IsIntTy());
+    // Zero sized integers are not usually valid. However, during semantic
+    // analysis when we're inferring types, the first pass may not have
+    // enough information to figure out the exact size of the integer. Later
+    // passes infer the exact size.
+    assert(bits == 0 || bits == 1 || bits == 8 || bits == 16 || bits == 32 ||
+           bits == 64 || bits == 128);
+    size_bits_ = bits;
```

Comment:
Could we put the 128-bit integer truncation in here? That should guarantee that we catch every place an integer's size is set

Reply:
I've updated the code accordingly 🙂 
However, we currently don't handle multiple 128-bits arguments correctly. See the commented-out `EXPECT` in `tests/runtime/dwarf`'s 128-bits tests.

Reply:
I've documented the issue in #3236.